bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/EntsoeHandler.java

```diff
@@ -0,0 +1,285 @@
+import org.openhab.core.library.unit.CurrencyUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.TimeSeries;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EntsoeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jørgen Melhus - Initial contribution
+ */
+@NonNullByDefault
+public class EntsoeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EntsoeHandler.class);
+
+    private EntsoeConfiguration config;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private Unit<Currency> baseUnit = CurrencyUnits.BASE_CURRENCY;
+    private Unit<Currency> fromUnit = new CurrencyUnit(EntsoeBindingConstants.ENTSOE_CURRENCY, null);
+
+    private Map<ZonedDateTime, Double> responseMap;
+
+    // Use region-based ZoneId instead of CET to handle transition to CEST automatically
+    private final ZoneId cetZoneId = ZoneId.of("Europe/Paris");
+    private final ZoneId utcZoneId = ZoneId.of("UTC");
+
+    private ZonedDateTime lastDayAheadReceived = ZonedDateTime.of(LocalDateTime.MIN, cetZoneId);
+
+    private int historicDaysInitially = 0;
+
+    public EntsoeHandler(Thing thing) {
+        super(thing);
+        config = getConfigAs(EntsoeConfiguration.class);
+        responseMap = new TreeMap<>();
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        logger.trace("channelLinked(channelUID:{})", channelUID.getAsString());
+        String channelID = channelUID.getId();
+
+        if (EntsoeBindingConstants.CHANNEL_SPOT_PRICE.equals(channelID)) {
+            if (responseMap.isEmpty()) {
+                refreshPrices();
+            }
+            updateCurrentHourState(EntsoeBindingConstants.CHANNEL_SPOT_PRICE);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+        super.dispose();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("handleCommand(channelUID:{}, command:{})", channelUID.getAsString(), command.toFullString());
+
+        if (command instanceof RefreshType) {
+            fetchNewPrices();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(EntsoeConfiguration.class);
+
+        if (historicDaysInitially == 0) {
+            historicDaysInitially = config.historicDays;
+        }
+        updateStatus(ThingStatus.UNKNOWN);
```

Comment:
Since polling is skipped when having no linked items, status will never be updated to **ONLINE**.
```suggestion
        if (isLinked(EntsoeBindingConstants.CHANNEL_SPOT_PRICE)) {
            updateStatus(ThingStatus.UNKNOWN);
        } else {
            updateStatus(ThingStatus.ONLINE);
        }
```